nvflare/edge/edge_job.py

```diff
@@ -0,0 +1,147 @@
+
+        check_object_type("assessor", assessor, Assessor)
+        check_positive_int("num_rounds", num_rounds)
+        check_str("task_name", task_name)
+
+        assessor_id = self.to_server(assessor, id="wf_assessor")
+
+        controller = ScatterAndGatherForEdge(
+            assessor_id=assessor_id,
+            num_rounds=num_rounds,
+            task_name=task_name,
+            task_check_period=0.5,
+            assess_interval=assess_interval,
+            update_interval=update_interval,
+        )
+
+        self.to_server(controller, id="sage")
+        self.server_config_added = True
+
+    def configure_client(
+        self,
+        aggregator_factory: AggregatorFactory,
+        max_model_versions: int,
+        update_timeout=5.0,
+        executor_task_name="train",
+        simulation_config_file: str = None,
+    ):
+        """Set up client config.
+
+        Args:
+            aggregator_factory: an AggregatorFactory object to create aggregators when needed.
+            max_model_versions: max number of model versions to keep.
+            update_timeout: timeout for status update messages.
+            executor_task_name: task name for executor.
+            simulation_config_file: config file for local simulation (optional).
+
+        Returns: None
+
+        """
+        if self.client_config_added:
+            raise RuntimeError("client config is already added")
+
+        check_object_type("aggregator_factory", aggregator_factory, AggregatorFactory)
+        check_positive_int("max_model_versions", max_model_versions)
+        check_positive_number("update_timeout", update_timeout)
+        check_str("executor_task_name", executor_task_name)
+
+        if simulation_config_file:
+            check_str("simulation_config_file", simulation_config_file)
+
+        self.to_clients(EdgeTaskReceiver(), id="edge_task_receiver")
+
+        aggr_factory_id = self.to_clients(aggregator_factory, id="aggr_factory")
+        executor = EdgeModelExecutor(
+            aggr_factory_id=aggr_factory_id, max_model_versions=max_model_versions, update_timeout=update_timeout
+        )
+        self.to_clients(executor, id="executor", tasks=[executor_task_name])
+
+        if simulation_config_file:
+            if not os.path.isfile(simulation_config_file):
+                raise ValueError(f"file {simulation_config_file} does not exist or is not a valid file")
+
+            try:
+                with open(simulation_config_file, "r") as f:
+                    json.load(f)
+            except Exception as ex:
+                raise ValueError(f"file {simulation_config_file} is not a valid JSON file: {ex}")
+
+            self.to_clients(FileSource(simulation_config_file, app_folder_type="config"))
+
+            base_name = os.path.basename(simulation_config_file)
+            conf_file = "{JOB_CONFIG_DIR}/" + f"{base_name}"
+            self.to_clients(TPRunner(conf_file), id="tp_runner")
```

Comment:
as this defaults to TP runner, shall we call it edge_tp_job to avoid confusion with other simulation modes?
(and does it mean we "inexplictly hide" device_runner?)